### PR TITLE
add node.js 4 (current LTS version)

### PR DESCRIPTION
**Pull request by robotanarchy**
I've kept the other nodejs recipe, because v4.x and higher require a newer GCC
than 4.7, which gets built by default in sabotage.

Also I've added an extra check for the GCC version at the very start of the
build. Otherwise the configure script would only print a notice, and the build
fails somehwere in the middle without throwing an error related to the GCC version.

Built successfully with GCC 5.2.0 on my machine.

### CODE REVIEW

File: pkg/nodejs4 [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+filesize=12791572
+sha512=fe6563ca641581afa4ba7fda1b25175eae474aeb196504cd82a6486bf9dfe6c753c554b31d9aa3b6453c509aa792615e897343de69a0b8bfd26759efaea2e1cb
+
+[deps.host]
+dynamic-toolchain
+python
+
+[build]
+# Better fail here, than in the middle of the build!
+if ! (gcc --version | grep -q -E '((4\.[8-9])|([5-9]\.[0-9]+))\.'); then
```
Comment (rofl0r) [inline pkg/nodejs4]: please use `$CC` rather than gcc, because on x-compiles you test only for the host compiler

File: pkg/nodejs4 [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	echo "ERROR: GCC 4.8 or higher required!"
+	exit 1
+fi
+
+patch -p1 < "$K"/nodejs-nameser_compat.h
+patch -p1 < "$K"/nodejs-openssl_termios.h
+patch -p1 < "$K"/nodejs-prefix.patch
+
+
+CFLAGS="-D_GNU_SOURCE" CXXFLAGS=-D_GNU_SOURCE ./configure --prefix="$butch_prefix"
```
Comment (rofl0r) [inline pkg/nodejs4]: would you mind changing that to 
```
CFLAGS="-D_GNU_SOURCE -$optcflags" CXXFLAGS="-D_GNU_SOURCE $optcflags" LDFLAGS="$optldflags"
```
please ? that way user's cflags get used

File: pkg/nodejs4 [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# the python configure script does not care about C(XX)FLAGS being passed
+sed -i "s@cflags': \[\]@cflags': ['-D_GNU_SOURCE -D_BSD_SOURCE']@" config.gypi
```
Comment (rofl0r) [inline pkg/nodejs4]: same here

File: pkg/nodejs [hunk 4]
```diff
@@ -10,11 +10,14 @@ dynamic-toolchain
...
 [build]
+# NOTE: node.js 0.12 is not maintained anymore upstream, but it is the
+# last version that builds with GCC 4.7.
+
 patch -p1 < "$K"/nodejs-nameser_compat.h
 patch -p1 < "$K"/nodejs-openssl_termios.h
 patch -p1 < "$K"/nodejs-prefix.patch
 
-CFLAGS="-D_GNU_SOURCE" CXXFLAGS=-D_GNU_SOURCE ./configure --prefix="$butch_prefix"
+CFLAGS="-D_GNU_SOURCE -$optcflags" CXXFLAGS="-D_GNU_SOURCE $optcflags" LDFLAGS="$optldflags" ./configure --prefix="$butch_prefix"
```
Comment (rofl0r) [inline pkg/nodejs]: this "-" before optcflags likely breaks things

**Comment by robotanarchy** [comment]
@rofl0r : fixed. as a bonus, bumped the packages to the current version to fix [CVE-2015-8027 Denial of Service Vulnerability](https://nodejs.org/en/blog/vulnerability/december-2015-security-releases/). Tested both versions, they compile and work fine on my machine™.

**Comment by rofl0r** [comment]
merged via 39d1fd2

**Pull request closed without merging**